Moved airflow url to helper, added env var

cwurtz opened the pull request:
Moved the airflow url to a helper since it is used in multiple places.  Added env var for the webserver base_url.

resolves astronomerio/astronomer-ee/74

schnie commented on charts/airflow/templates/ingress.yaml:
```diff
@@ -30,13 +30,14 @@ spec:
   {{- end }}
   rules:
-    - host: airflow.{{ .Values.global.baseDomain }}
+    - host:{{- include "airflow_url" . }}
```
@cwurtz can we add a space here too?

schnie commented on charts/airflow/templates/_helpers.yaml:
```diff
@@ -43,3 +46,7 @@
 {{- end }}
+
+# On the same line due to whitespace issue
+{{ define "airflow_url" }} airflow.{{ .Values.global.baseDomain }}
```
@cwurtz could we put this on multiple lines and use `indent` to control spacing?

schnie commented on charts/airflow/templates/_helpers.yaml:
```diff
@@ -16,6 +16,9 @@
   - name: ASTRONOMER__AIRFLOW__WORKER_LOG_RETENTION
     value: "15"
+  - name: AIRFLOW__WEBSERVER__BASE_URL
+    value:{{- include "airflow_url" . }}
```
@cwurtz can we add a space after the colon here too?

schnie commented:
@cwurtz @andscoop should we merge this for the release? Could rebuild the release, or release 0.2.1.

cwurtz commented:
I'd rather update the 0.2.0 release rather than push out 0.2.1.  However, I think this could also wait a week or two and be included in a larger release.  Up to you though

schnie commented:
@cwurtz looks like it might need to be updated to fit with the new domain name formats, using a hypen rather than a subdomain.

Why did we do that again? I remember talking about it but forget the reason.

cwurtz commented:
Err, yeah need to update the format now that 0.2.0 is cut.

We're doing the hyphen to allow for multiple airflow instances.  Each will have their set of urls:
[release-name]-airflow.baseDomain
[release-name]-flower.baseDomain
... etc

We could do another layer of domains, however then you'd have to either be forced to use let's encrypt, or buy/configure multiple wildcard certs

schnie commented:
Ah, that's right. Makes sense.

On Tue, May 8, 2018 at 11:04 PM Courtney Wurtz <notifications@github.com>
wrote:

> Err, yeah need to update the format now that 0.2.0 is cut.
>
> We're doing the hyphen to allow for multiple airflow instances. Each will
> have their set of urls:
> [release-name]-airflow.baseDomain
> [release-name]-flower.baseDomain
> ... etc
>
> We could do another layer of domains, however then you'd have to either be
> forced to use let's encrypt, or buy/configure multiple wildcard certs
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/astronomerio/helm.astronomer.io/pull/25#issuecomment-387605935>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA_Qwm_tP9-UfizWP60ciDVomHnnBDF1ks5twlyhgaJpZM4TqpJo>
> .
>


-- 
*Greg Neiheisel* / CTO Astronomer.io

andscoop commented:
LGTM

cwurtz commented:
Updated the commit to reflect the new release-name prefix for 0.2.0.  Also moved the rest of the urls to a helper as it seems out of place with 1/4 being a helper include
